File: leetcode3/남효정/4000. Largest Integer With Given Digit Sum.py
class Solution:
    def largestInteger(self, n: int, s: int) -> int:
        if s == 0: 
            return 0

        if n * 9 < s:
            return -1
        
        div = s // 9
        mod = s % 9

        if mod == 0:
            return int('9' * div + (n - div) * '0')

        return int('9' * div + str(mod) + (n - div - 1) * '0')
            
        

File: leetcode3/염혜정/4000. Largest Integer With Given Digit Sum.java
// 9로 나눠가며 9부터 채우고 나머지 값 붙이기
// 무조건 n차수 맞추기
// O(n)

class Solution {
    public int largestInteger(int n, int s) {
        if (s==0) return 0;
        if (s > n*9) return -1;

        StringBuilder sb = new StringBuilder();
        int cnt_9 = s / 9;
        int r = s % 9;
        for (int i = 0; i<cnt_9; i++) {
            sb.append("9");
        }

        int digitCnt = cnt_9;
        if (r != 0) {
            sb.append(r);
            digitCnt++;
        }

        for (int i = digitCnt; i<n; i++) {
            sb.append("0");
        }

        int result = Integer.parseInt(sb.toString());
        return result;
    }
}
